src/pages/library/datasetBuilder/CohortEditor.ts

```diff
@@ -0,0 +1,428 @@
+import { datasetBuilderCohorts } from 'src/pages/library/datasetBuilder/state';
+
+type CriteriaViewProps = { criteria: AnyCriteria; deleteCriteria: (criteria: AnyCriteria) => void };
+
+const CriteriaView = ({ criteria, deleteCriteria }: CriteriaViewProps) => {
+  return div(
+    {
+      style: {
+        display: 'flex',
+        width: '100%',
+        justifyContent: 'space-between',
+        alignItems: 'baseline',
+        paddingBottom: 5,
+        marginTop: 10,
+        marginBottom: 5,
+        borderBottom: `1px solid ${colors.dark(0.35)}`,
+      },
+    },
+    [
+      div({ style: { margin: '5px', display: 'flex', alignItems: 'center' } }, [
+        h(
+          Link,
+          {
+            'aria-label': 'delete criteria',
+            onClick: () => {
+              deleteCriteria(criteria);
+            },
+          },
+          [icon('minus-circle', { size: 24, style: { color: colors.danger() } })]
+        ),
+        div({ style: { marginLeft: 5 } }, [
+          Utils.cond(
+            [
+              'domainType' in criteria,
+              () => {
+                const domainCriteria = criteria as DomainCriteria;
+                return h(Fragment, [strong([`${domainCriteria.domainType.category}:`]), ` ${domainCriteria.name}`]);
+              },
+            ],
+            [
+              'listType' in criteria,
+              () => {
+                const listCriteria = criteria as ProgramDataListCriteria;
+                return h(Fragment, [strong([`${criteria.name}:`]), ` ${listCriteria.value.name}`]);
+              },
+            ],
+            [
+              'rangeType' in criteria,
+              () => {
+                const rangeCriteria = criteria as ProgramDataRangeCriteria;
+                return h(Fragment, [strong([`${criteria.name}:`]), ` ${rangeCriteria.low} - ${rangeCriteria.high}`]);
+              },
+            ],
+            [Utils.DEFAULT, () => div(['Unknown criteria type'])]
+          ),
+        ]),
+      ]),
+      `Count: ${criteria.count}`,
+    ]
+  );
+};
+
+export const renderCriteriaView = (deleteCriteria: (criteria: AnyCriteria) => void) => (criteria: AnyCriteria) =>
+  h(CriteriaView, { deleteCriteria, criteria, key: criteria.id });
+
+let criteriaCount = 1;
+const selectDomainCriteria = (domainType: DomainType): DomainCriteria => {
+  // This needs to be replaced with a UI that lets users select the criteria they want from
+  // the list of concepts for this domain.
+  return {
+    domainType,
+    name: domainType.values[0],
+    id: criteriaCount++,
+    // Need to call the API service to get the count for this criteria.
+    count: 100,
+  };
+};
+
+const createDefaultListCriteria = (listType: ProgramDataListType): ProgramDataListCriteria => {
+  return {
+    listType,
+    name: listType.name,
+    id: criteriaCount++,
+    count: 100,
+    value: listType.values[0],
```

Comment:
With this as multiselect, should this be `value: [listType.values[0]]`?

Reply:
Yes, although right now there's no select control at all. Even this much state may have been too much "coding ahead" since I'm not sure exactly what will be needed for the UI controls.. so I think I'll push this off until we add the inline editor.